service/src/main/java/org/cbioportal/genome_nexus/service/enricher/SignalAnnotationEnricher.java

```diff
@@ -23,13 +24,9 @@ public SignalAnnotationEnricher(
     @Override
     public void enrich(VariantAnnotation annotation) {
         SignalAnnotation signalAnnotation = new SignalAnnotation();
-        String hgvsg = annotation.getHgvsg();
-
-        if (hgvsg != null) {
-            List<SignalMutation> mutations = this.signalMutationService.getSignalMutationsByHgvsg(hgvsg);
-            signalAnnotation.setAnnotation(mutations);
-        }
-
+        GenomicLocation genomicLocation = GenomicLocationResolver.resolve(annotation);
```

Comment:
Let's revert the changes in `GenomicLocationResolver` and create an instance of `GenomicLocationResolver` in the constructor instead. Something like this should work:

```Java
private final GenomicLocationResolver genomicLocationResolver;

public SignalAnnotationEnricher(...) {
    ...
    this.genomicLocationResolver = new GenomicLocationResolver()
}

```

Then you can simply do this here:
```suggestion
        GenomicLocation genomicLocation = this.genomicLocationResolver.resolve(annotation);
```

Reply:
@onursumer Sure, updated. Why static is not good here?

Reply:
Thanks! Static methods may break component autowiring. Also, I think only dynamic methods can be mocked in test classes.